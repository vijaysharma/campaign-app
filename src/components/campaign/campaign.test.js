import React from "react";
import ReactDOM from "react-dom";
import Campaign from "./campaign";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

const handleAction = jest.fn(() => true);
afterEach(cleanup);
it("Campaign renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Campaign
      campaign={{
        id: 1,
        title: "Emails",
        history: [{ title: "Created", on: "2:54 pm", by: "Anonymous" }],
        selected: true
      }}
      onPause={handleAction}
      onToggleComment={handleAction}
      onComment={handleAction}
      onRename={handleAction}
      onDelete={handleAction}
      onSelect={handleAction}
      listNumber={1}
    ></Campaign>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
it("Campaign renders correctly", () => {
  const { getByTestId } = render(
    <Campaign
      campaign={{
        id: 1,
        title: "Emails",
        history: [{ title: "Created", on: "2:54 pm", by: "Anonymous" }],
        selected: true
      }}
      onPause={handleAction}
      onToggleComment={handleAction}
      onComment={handleAction}
      onRename={handleAction}
      onDelete={handleAction}
      onSelect={handleAction}
      listNumber={1}
    ></Campaign>
  );
  expect(getByTestId("campaign")).toHaveTextContent(
    "1Campaign 1 - EmailsCreated at 2:54 pmPauseCommentRenameDelete"
  );
});

it("Campaign matches snapshot", () => {
  const campaignSnap = renderer
    .create(
      <Campaign
        campaign={{
          id: 1,
          title: "Emails",
          history: [{ title: "Created", on: "2:54 pm", by: "Anonymous" }],
          selected: true
        }}
        onPause={handleAction}
        onToggleComment={handleAction}
        onComment={handleAction}
        onRename={handleAction}
        onDelete={handleAction}
        onSelect={handleAction}
        listNumber={1}
      ></Campaign>
    )
    .toJSON();
  expect(campaignSnap).toMatchSnapshot();
});
