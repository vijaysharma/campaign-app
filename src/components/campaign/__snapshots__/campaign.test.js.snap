// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign matches snapshot 1`] = `
<li
  className="list-group-item campaign active"
  data-testid="campaign"
  onClick={[Function]}
>
  <div
    className="campaign-details"
  >
    <span
      className="campaign-list-number"
    >
      1
    </span>
    <div
      className="campaign-title"
    >
      <h5>
        Campaign 
        1
         -
         
        <span>
          Emails
        </span>
      </h5>
      <span
        className="camapaign-list-created"
      >
        Created at
         
        2:54 pm
      </span>
    </div>
  </div>
  <div
    className="campaign-actions"
  >
    <button
      data-testid="actionButton"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-pause-circle fa-w-16 "
        data-icon="pause-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm-16 328c0 8.8-7.2 16-16 16h-48c-8.8 0-16-7.2-16-16V176c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16v160zm112 0c0 8.8-7.2 16-16 16h-48c-8.8 0-16-7.2-16-16V176c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16v160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Pause
      </span>
    </button>
    <button
      data-testid="actionButton"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-comment fa-w-16 "
        data-icon="comment"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Comment
      </span>
    </button>
    <button
      data-testid="actionButton"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-edit fa-w-18 "
        data-icon="edit"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Rename
      </span>
    </button>
    <button
      data-testid="actionButton"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-trash fa-w-14 "
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Delete
      </span>
    </button>
  </div>
</li>
`;
